iot_input/controller/iot_input_controller.py

```diff
@@ -2,23 +2,41 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 import json
-from odoo import http
+from odoo import http, _
 
 
 class CallIot(http.Controller):
     @http.route([
         '/iot/<serial>/action',
-        ], type='http', auth="none", methods=['POST'], csrf=False)
+    ], type='http', auth="none", methods=['POST'], csrf=False)
     def call_unauthorized_iot(self, serial, *args, **kwargs):
         request = http.request
         if not request.env:
             return json.dumps(False)
         return json.dumps(request.env['iot.device.input'].sudo().get_device(
             serial, kwargs['passphrase']).call_device(kwargs['value']))
 
+    @http.route([
+        '/iot/<serial>/action2',
+    ], type='http', auth="none", methods=['POST'], csrf=False)
+    def call_unauthorized_iot_action(self, serial, *args, **kwargs):
+        request = http.request
+        if not request.env:
+            return json.dumps({'status': 'error',
+                               'message': _('env not set')})
+        if 'passphrase' not in kwargs:
+            return json.dumps({'status': 'error',
+                               'message': _('Passphrase is required')})
+        if 'value' not in kwargs:
+            return json.dumps({'status': 'error',
+                               'message': _('Value is required')})
+        return json.dumps(
```

Comment:
maybe you could add something like a try... except and write on the logging the traceback and allways returna  `{'status': 'error'}` message

Reply:
I don't get what you mean here. Could you elaborate a bit?

Reply:
https://github.com/tegin/iot/commit/2ad45887cd10f3d0314959fdaef5ecd0735e9c1c

Reply:
This allows to simplify the code, as only one code for execution is required

Reply:
But won't that hide the error cause from the IoT device?
My intention was to provide better error reporting to the IoT device, because in an industry 4.0 environment that could mean several meters away from the Odoo server and possibly standalone.

Reply:
Well, this will show a generic message when an error ocurred, but, we could add the traceback if you think this is better